File: auth/authentication/core/src/main/java/io/ehdev/conrad/auth/cookie/UserCookieMangerImpl.java
package io.ehdev.conrad.auth.cookie;

import org.springframework.stereotype.Service;
import org.springframework.web.util.CookieGenerator;

import javax.servlet.http.Cookie;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

@Service
public class UserCookieMangerImpl implements UserCookieManger {

    static final String COOKIE_NAME = "conrad_cookie";

    private final CookieGenerator userCookieGenerator = new CookieGenerator();

    public UserCookieMangerImpl() {
        this.userCookieGenerator.setCookieName(COOKIE_NAME);
    }

    @Override
    public void addCookie(String contents, HttpServletResponse response) {
        userCookieGenerator.addCookie(response, contents);
    }

    @Override
    public void removeCookie(HttpServletResponse response) {
        userCookieGenerator.addCookie(response, "");
    }

    @Override
    public String readCookieValue(HttpServletRequest request) {
        Cookie[] cookies = request.getCookies();
        if (cookies == null) {
            return null;
        }
        for (Cookie cookie : cookies) {
            if (cookie.getName().equals(userCookieGenerator.getCookieName())) {
                return cookie.getValue();
            }
        }
        return null;
    }

}

File: auth/authentication/src/main/kotlin/tech/crom/security/authentication/cookie/UserCookieMangerImpl.kt
package tech.crom.security.authentication.cookie

import org.springframework.stereotype.Service
import org.springframework.web.util.CookieGenerator
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

@Service
class UserCookieMangerImpl : UserCookieManger {

    private val userCookieGenerator = CookieGenerator()

    init {
        this.userCookieGenerator.cookieName = "crom_cookie"
    }

    override fun addCookie(contents: String, response: HttpServletResponse) {
        userCookieGenerator.addCookie(response, contents)
    }

    override fun removeCookie(response: HttpServletResponse) {
        userCookieGenerator.addCookie(response, "")
    }

    override fun readCookieValue(request: HttpServletRequest): String? {
        val cookies = request.cookies ?: return null
        for (cookie in cookies) {
            if (cookie.name == userCookieGenerator.cookieName) {
                return cookie.value
            }
        }
        return null
    }
}
